zeppelin-server/src/test/java/org/apache/zeppelin/integration/TestSparkParagraph.java

```diff
@@ -0,0 +1,238 @@
+import org.apache.zeppelin.WebDriverManager;
+import org.hamcrest.CoreMatchers;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ErrorCollector;
+import org.openqa.selenium.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class TestSparkParagraph extends AbstractZeppelinIT {
+
+  private static final Logger LOG = LoggerFactory.getLogger(TestSparkParagraph.class);
+
+  @Rule
+  public ErrorCollector collector = new ErrorCollector();
+
+  @Before
+  public void startUp() {
+    if (!endToEndTestEnabled()) {
+      return;
+    }
+    driver = WebDriverManager.getWebDriver();
+  }
+
+  @After
+  public void tearDown() {
+    if (!endToEndTestEnabled()) {
+      return;
+    }
+
+    driver.quit();
+  }
+
+  @Test
+  public void testSpark() throws InterruptedException {
+    if (!endToEndTestEnabled()) {
+      return;
+    }
+    try {
+      createNewNote();
+
+      waitForParagraph(1, "READY");
+
+      WebElement paragraph1Editor = driver.findElement(By.xpath(getParagraphXPath(1) + "//textarea"));
+
+      paragraph1Editor.sendKeys("import org.apache.commons.io.IOUtils"
+          + Keys.ENTER +
+
+          "import java.net.URL" + Keys.ENTER +
+
+          "import java.nio.charset.Charset" + Keys.ENTER +
+
+          "val bankText = sc.parallelize" + Keys.chord(Keys.SHIFT, "9") +
+          "IOUtils.toString" + Keys.chord(Keys.SHIFT, "9") + "new URL" + Keys.chord(Keys.SHIFT, "9")
+          + "\"https://s3.amazonaws.com/apache" + Keys.SUBTRACT + "zeppelin/tutorial/bank/bank." +
+          "csv\"),Charset.forName" + Keys.chord(Keys.SHIFT, "9") + "\"utf8\"))" +
+          ".split" + Keys.chord(Keys.SHIFT, "9") + "\"\\n\"))" + Keys.ENTER +
+
+          "case class Bank" + Keys.chord(Keys.SHIFT, "9") +
+          "age: Integer, job: String, marital: String, education: String, balance: Integer)" +
+          Keys.ENTER +
+          Keys.ENTER +
+
+          "val bank = bankText.map" + Keys.chord(Keys.SHIFT, "9") + "s => s.split" +
+          Keys.chord(Keys.SHIFT, "9") + "\";\")).filter" + Keys.chord(Keys.SHIFT, "9") +
+          "s => s" + Keys.chord(Keys.SHIFT, "9") + "0) " + Keys.chord(Keys.SHIFT, "1") +
+          "= \"\\\"age\\\"\").map" + Keys.chord(Keys.SHIFT, "9") +
+          "s => Bank" + Keys.chord(Keys.SHIFT, "9") + "s" + Keys.chord(Keys.SHIFT, "9") +
+          "0).toInt,s" + Keys.chord(Keys.SHIFT, "9") + "1).replaceAll" +
+          Keys.chord(Keys.SHIFT, "9") + "\"\\\"\", \"\")," +
+          "s" + Keys.chord(Keys.SHIFT, "9") + "2).replaceAll" +
+          Keys.chord(Keys.SHIFT, "9") + "\"\\\"\", \"\")," +
+          "s" + Keys.chord(Keys.SHIFT, "9") + "3).replaceAll" +
+          Keys.chord(Keys.SHIFT, "9") + "\"\\\"\", \"\")," +
+          "s" + Keys.chord(Keys.SHIFT, "9") + "5).replaceAll" +
+          Keys.chord(Keys.SHIFT, "9") + "\"\\\"\", \"\").toInt" + ")" +
+          ").toDF" + Keys.chord(Keys.SHIFT, "9") + ")" + Keys.ENTER +
```

Comment:
Looks great, but how do you think, is it possible to improve this part a bit, to something more readable and easier to maintain?

May be extracting variables for `Keys.chord(Keys.SHIFT, "9")` and giving them a short but descriptive names like enter, or eol, etc could help